### DISCUSSION TITLE
StreamFX 0.12.0 Beta 164

### DISCUSSION THREAD

**Discussion by Xaymar** [Xaymar/obs-stream-effects#989] [Announcements]
# ⚠ Preview Release not for Production! ⚠
This release is meant for testing in isolated conditions and early adopters only, and should never be used in production. If you use one of these releases, all promises of support, warranty, functionality, and similar are immediately void and if things break you are on your own. [The latest production ready release can be found here](https://github.com/Xaymar/obs-StreamFX/releases/latest).

# Please read the [Installation Guide](https://github.com/Xaymar/obs-StreamFX/wiki/Installation) and System Requirements!
The installation of StreamFX requires that you follow certain instructions, as well as fulfill a minimum hardware and software requirement. Please ensure that you have read and verified both before asking for help, as many issues can be fixed by reading both of them in their entirety.

# Support StreamFX on Github Sponsors or Patreon!
[<img align="left" alt="GitHub Sponsors" src="https://user-images.githubusercontent.com/437395/106463119-48412400-6497-11eb-9934-baed8e0bdd73.png" height="70px"/>](https://github.com/sponsors/xaymar) [<img align="right" alt="Patreon" src="https://user-images.githubusercontent.com/437395/106462708-bd602980-6496-11eb-8f35-038577cf8fd7.jpg" height="70px"/>](https://patreon.com/xaymar) The StreamFX project is only possible through your support, either as a [translator](https://github.com/Xaymar/obs-StreamFX/issues/36) or as a Supporter on either [Patreon](https://patreon.com/xaymar) or [Github](https://github.com/sponsors/xaymar). Your support helps keep StreamFX free from Ads, even if it's just translating a few lines per month, or 1$ per month. If continuous support isn't your thing, maybe a one-time donation via [PayPal](https://paypal.me/xaymar) is?

# System Requirements
<table style="table-layout: fixed">
  <tr>
    <th width="5%"><img width=10000000px height=1 /></th>
    <th width="5%"></th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-windows.svg" alt="Windows"><br>Windows</th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-linux.svg" alt="Linux"><br>Linux</th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-macos.svg" alt="MacOS"><br>MacOS</th>
  </tr>
  <tr>
    <th rowspan=2>OS</th>
    <th>Minimum</th>
    <td>Windows 10 21H2 (or newer)</td>
    <td>Ubuntu 20.04 (or equivalent)<br>Ubuntu 22.04 (or equivalent)</td>
    <td>MacOS 10.15 (or newer)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td>Windows 10 21H2 (or newer)</td>
    <td>Ubuntu 20.04 (or equivalent)<br>Ubuntu 22.04 (or equivalent)</td>
    <td>MacOS 12 (or newer)</td>
  </tr>
  <tr>
    <th rowspan=4>CPU</th>
    <th rowspan=2>Minimum</th>
    <td colspan=3>
      <i>Any x86-64-v3 compatible:</i> <a href="https://en.wikipedia.org/wiki/Excavator_(microarchitecture)">AMD Excavator Series</a>, <a href="https://en.wikipedia.org/wiki/Haswell_(microarchitecture)">Intel Haswell Series</a><br/>
      <i>Required Instruction Sets:</i> x86-64-v2, AVX, BMI1, BMI2, F16C, FMA, MOVBE
    </td>
  </tr>
  <tr>
    <td colspan=2 style="display: none;"></td>
    <td>
      <i>Any ARM64-v8.5-A compatible:</i> <a href="https://en.wikipedia.org/wiki/Apple_M1">Apple M1 Series</a>
    </td>
  </tr>
  <tr>
    <th rowspan=2>Recommended</th>
    <td colspan=3>
      <i>Any x86-64-v3 compatible:</i> <a href="https://en.wikipedia.org/wiki/Zen_2">AMD Zen 2 Series</a>, <a href="https://en.wikipedia.org/wiki/Comet_Lake_(microprocessor)">Intel Comet Lake Series</a><br/>
      <i>Required Instruction Sets:</i> x86-64-v2, AVX, BMI1, BMI2, F16C, FMA, MOVBE
    </td>
  </tr>
  <tr>
    <td colspan=2 style="display: none;"></td>
    <td>
      <i>Any ARM64-v8.5-A compatible:</i> <a href="https://en.wikipedia.org/wiki/Apple_M1">Apple M1 Series</a>
    </td>
  </tr>
  <tr>
    <th rowspan=2>Memory (RAM)</th>
    <th>Minimum</th>
    <td colspan=3>4 GiB (free)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td>8 GiB (free)</td>
    <td colspan=2>6 GiB (free)</td>
  </tr>
  <tr>
    <th rowspan=2>Video (GPU)</th>
    <th>Minimum</th>
    <td colspan=3>
      DirectX 11 / OpenGL 3 capable, Shader Model 4, 4 GiB VRAM (total)
    </td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>
      DirectX 12 / OpenGL 4 capable, Shader Model 5, 6 GiB VRAM (total)
    </td>
  </tr>
  <tr>
    <th rowspan=2>Disk Space</th>
    <th>Minimum</th>
    <td colspan=3>~50 MiB (free)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>~50 MiB (free)</td>
  </tr>
  <tr>
    <th rowspan=2>OBS Version</th>
    <th>Minimum</th>
    <td colspan=3>29.0.0-beta2</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>29.0.0</td>
  </tr>
  <tr>
    <th colspan=2 center>Other</th>
    <td>
      <a href="https://aka.ms/vs/17/release/vc_redist.x64.exe">Microsoft Visual C/C++ Redistributable 2017-2022</a>
    </td>
    <td>
      Qt for obs-studio<br>
      FFmpeg for obs-studio<br>
      System Libraries must match chosen Ubuntu version!<br>
    </td>
    <td>
    </td>
  </tr>
</table>

# What's Changed
## New Release process: Alpha -> Beta -> Candidate -> Release!
The old two stage system didn't work out quite as I'd have hoped: It didn't communicate the dangers of Testing well, and did not differentiate between different stability levels. To solve this I've adopted a new release process now available in StreamFX, which follows the usual development cycle of Software itself: Alpha -> Beta -> Candidate -> Release. The further right you go in that graph, the more stable software is supposed to be, at the cost of new features and changes. The further left you go, the more up-to-date things will be, but you sacrifice stability and safety.

### Shader Texture Inputs no longer corrupt Scene Collections
For quite some time, Shader Texture Inputs were haunted by ghosts from the past, also known as work-arounds and deprecated code. This resulted in Shader Texture Inputs keeping stale references around and requiring an OBS Studio restart to actually update. Several code modernization cycles later, this is now fixed and Shader Texture Inputs immediately update as expected - while also no longer keeping stale references around, so your Scene Collections are safe again.

### Dynamic Mask no longer self-references and double-frees its Input mask
This should have gone up in flames way sooner than it did, and yet it was almost completely invisible until OBS Studio 27.1. The bug dates as far back as [v0.7](https://github.com/Xaymar/obs-StreamFX/releases/tag/0.7.0) and is pretty severe: The Dynamic Mask filter would increment their own reference count every frame if no *Input* was selected, and would double-free the Input without clearing the pointer to it. And somehow, this didn't explode until now.

### Loadable Bundles on MacOS
With OBS Studio 28.0 came the introduction of Loadable Bundles, a variation of MacOS's Bundle format. This format offers a lot of advantages for almost no drawbacks, and StreamFX from now on uses this packaging format on MacOS. As a bonus, FFmpeg Encoders should now be available on MacOS too.

#### ⚠ This change requires that you cleanly remove old versions of StreamFX first!
As the packaging and installation method has changed, please follow the [Uninstallation guide](https://github.com/Xaymar/obs-StreamFX/wiki/Uninstallation) to clean up any left over old files. Failure to do so will result in undefined behavior and may render OBS Studio entirely unusable. **You have been warned!**

### Improved System-wide installation on Windows
Investigation into how OBS Studio actually loads plugins revealed a change that had gone under the radar entirely. Since [OBS Studio v0.15.0](https://github.com/obsproject/obs-studio/releases/tag/0.15.0) it is possible to load plugins from `%ProgramData%\obs-studio\plugins`, which is a much safer option and now the only supported System-wide installation mode for StreamFX. This functionality has been so hidden that even the official [obs-plugintemplate](https://github.com/obsproject/obs-plugintemplate/blob/056422b2614fed1037bc1f19e0e468165e3068ea/cmake/bundle/windows/installer-Windows.iss.in) doesn't make use of it. An added bonus is that this should support Steam users of OBS Studio.

### User-only installation on Windows
StreamFX can now be installed for individual users, instead of installing for everyone on the system! This is possible thanks to [a PR](https://github.com/obsproject/obs-studio/pull/4067) by @abcdw, which allows us to set two environment variable to install and load plugins from `%LocalAppData%\Programs\obs-studio\plugins` in addition to all the other locations. These installations will conflict with a System-wide install and as such can't coexist, but may still work with the Steam release of OBS Studio.

### Avid DNxHR Encoder (via FFmpeg) by @IceStormNG
A new contender for the visually lossless intermediate codec arrives: Avid DNxHR! Thanks to work done by @IceStormNG, it is now possible to use this encoder without configuration trickery and directly configure it with a proper user interface. As this encoder is based on the real codec specification, it may perform better than the Apple ProRes encoder while being way more compatible with video editing software.

### Split Framerate support for FFmpeg Encoders
All FFmpeg Encoders now support encoding a different framerate than OBS Studio itself is running at. This allows for streaming at 30 FPS while recording at 60 FPS, or even recording at 120 FPS, or other strange things. Note that this is quite literally breaking several specifications, so use at your own risk - if things break, be happy that at least they broke with a noticable impact!

### Universal Binaries for x86 & ARM64 support on MacOS
Users on ARM64 devices can now enjoy native ARM64 binaries for less CPU usage, and as such more overall performance on CPU focused tasks. While StreamFX is mostly GPU workloads, some of the internal code does actually do very heavy lifting that benefits from this.

### Initial sRGB and HDR support for Sources/Filters/Transitions
With HDR support in OBS Studio finally being available, it was necessary to finally tackle the issue of sRGB and HDR support once and for all. Many Sources/Filters/Transitions will get proper sRGB and HDR support over future releases to catch up with OBS Studio. You can check the *Version Information* on the wiki to see if a certain feature supports sRGB or HDR yet.

### Deprecation of the AMD AMF (via FFmpeg) encoders
With OBS Studio 28.0 the old obs-amd-encoder project is finally laid to rest and a new AMD AMF integration is provided. This AMD AMF integration is technically and functionally superior to both the old obs-amd-encoder and FFmpegs AMD AMF integration. All users are urged to migrate as soon as possible, as no guarantees are made for how long this deprecated encoder will remain. Additionally, users on Linux should put their focus on the open Pull Requests for Linux support ([1](https://github.com/obsproject/obs-studio/pull/7206), [2](https://github.com/obsproject/obs-studio/pull/4974)).

### Deprecation of the AV1 (via AOM) Encoder
With OBS Studio 27.2.4, the SVT-AV1 and AOM-AV1 encoders were integrated into OBS Studio directly. This integration is functionally identical, and as such replaces the StreamFX provided AOM-AV1 encoder. All users are urged to migrate as soon as possible, as no guaratees are made for how long this deprecated encoder will remain.

## [Detailed Changelog](https://github.com/Xaymar/obs-StreamFX/compare/0.11.0...0.12.0b164)

<details><summary>0.12.0b164</summary>

> * plugin: Only initialize GLAD on OpenGL graphics backend by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/983
> * project: Improve templates and UX by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/984
> * Code Cleanup by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/985
> * gfx/blur/box: Initialize streamfx::gfx::util by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/986
> * #818 Don't self-reference when masking is enabled by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/987
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/959
> * Fix CI and update Third Party code by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/988
</details>

<details><summary>0.12.0a151</summary>

> * Implement dynamically scaled Thread Pool by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/933
> * Save configuration after ~100ms have passed by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/934
> * templates/windows/installer: Don't create Icons in Portable mode by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/936
> * templates/windows/installer: Implement `/Portable` installer flag for CLI users by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/935
> * cmake, templates: Remove code signing integration by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/938
> * Don't show path selection for System & User mode by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/937
> * Fix some Use-After-Free issues with the D3D11 integration with FFmpeg by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/953
> * encoder/ffmpeg: Remove usage of `avcodec_close` by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/952
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/932
> * Implement support for Color Spaces by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/968
> * Dynamic Mask: Implement sRGB and HDR support, and fix #819 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/967
</details>

<details><summary>0.12.0a134</summary>

> * cmake: Require nlohmann-json when building the front-end by @rmnvgr in https://github.com/Xaymar/obs-StreamFX/pull/914
> * encoder/ffmpeg: Assume extra data and sei are present on first frame by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/919
> * encoder/ffmpeg: Add split framerate with integer fractions by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/921
> * encoder/ffmpeg: Improve software scaling quality by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/925
> * Add support for HDR formats by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/922
> * encoder/handler/prores_aw: Don't enable Key-Frame options by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/926
> * encoder/handler/dnxhd: Use endian-independent formats by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/927
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/915
> * Improve Windows installer with proper System install location, Current User install method, and better UI by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/930
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/928
</details>

<details><summary>0.12.0a117</summary>

> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/907
> * Improve Windows and Mac bundling by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/912
</details>

<details><summary>0.12.0a106</summary>

> * #881: Texture inputs to Shaders are unstable by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/884
> * Deprecate AMD AMF encoders by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/885
> * Deprecate AOM AV1 encoder by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/886
> * Add missing deprecation notices by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/889
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/867
> * cmake: Fix bad version strings by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/887
> * project: Updating contributing guidelines by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/894
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/890
> * project: Updating contributing guidelines by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/895
> * Fix a huge number of warnings by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/893
> * third-party: Update obs-studio to 28.0.0 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/901
> * code: Fix GCC warning diagnostic pragmas by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/902
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/897
</details>

<details><summary>0.12.0a77</summary>

> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/817
> * Improve GitHub integration by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/841
> * project: Fix incorrect tabs in YAML by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/842
> * Improve metadata generation by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/837
> * cmake: Add support for Qt6 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/866
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/832
> * Further improve CI and Github integration by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/843
> * ci: Remove Code Signing with self-signed certificate by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/873
> * Fix compilation errors with Debug and/or Profiling by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/876
> * Deduplicate CI build pipeline by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/874
> * Remove Auto-Dependency System by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/870
> * Working on source code [Refactor] by @GermanAizek in https://github.com/Xaymar/obs-StreamFX/pull/836
> * obs/gs/effect: Revert #836 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/880
> * Update libOBS to 28.0.0-rc1 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/878
</details>

<details><summary>0.12.0a45</summary>

> * Improve CMake build script(s) by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/813
> * ci: Add GCC back to the build by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/820
> * Add ClangCL, and update Windows SDK requirements by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/821
> * ci: Allow failures for experimental CI tests by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/824
> * ci: Use specific LLVM/Clang versions by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/825
> * project: Add support for multi stage release cycles by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/823
</details>

<details><summary>0.12.0a1</summary>

> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/778
> * ffmpeg-encoder/dnxhr: Added Avid DNxHR Encoder based on FFmpeg by @IceStormNG in https://github.com/Xaymar/obs-StreamFX/pull/784
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/786
> * Refactor C++ wrappers for libOBS by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/806
</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/Xaymar/obs-StreamFX/releases/tag/0.12.0b164'>StreamFX 0.12.0 Beta 164</a>.</em>

**Comment by ghost** [comment]
Do not install. Even after uninstalling the previous version, prior to installing this one, all features are missing.

**Reply by Wryyyong** [reply]
Read.

![obs](https://user-images.githubusercontent.com/28753317/205294899-bcb18f58-bd11-4263-9d5b-0d33f948eb48.png)

**Comment by ghost** [comment]
Please update main page.
Unsure how to reply to other message.
![image](https://user-images.githubusercontent.com/95929101/205413248-6285e633-fec0-4942-b17c-df6428aa42a6.png)

**Reply by Xaymar** [reply]
The main page is this GitHub. System Requirements are attached to each release. Failing to read the release notes is not my problem.

**Comment by DaeTheDerg** [comment]
My clean install of OBS crashes on launch when I use any version of this plugin past alpha 77 with error `Illegal Instruction (core dumped)`.
this is my log
```
$ obs -verbose --unfiltered-log
debug: Found portal inhibitor
Attempted path: share/obs/obs-studio/locale/en-US.ini
Attempted path: /usr/share/obs/obs-studio/locale/en-US.ini
Attempted path: share/obs/obs-studio/themes/Yami.qss
Attempted path: /usr/share/obs/obs-studio/themes/Yami.qss
info: Command Line Arguments: -verbose --unfiltered-log
info: Using EGL/X11
info: CPU Name: Intel(R) Xeon(R) CPU E5-1620 v2 @ 3.70GHz
info: CPU Speed: 3810.841MHz
info: Physical Cores: 4, Logical Cores: 8
info: Physical Memory: 32044MB Total, 20725MB Free
info: Kernel Version: Linux 6.0.11-arch1-1
info: Distribution: EndeavourOS Unknown
info: Session Type: x11
info: Window System: X11.0, Vendor: The X.Org Foundation, Version: 1.21.1
info: Qt Version: 6.4.1 (runtime), 6.4.1 (compiled)
info: Portable mode: false
Attempted path: share/obs/obs-studio/themes/Dark/no_sources.svg
Attempted path: /usr/share/obs/obs-studio/themes/Dark/no_sources.svg
qt.core.qmetaobject.connectslotsbyname: QMetaObject::connectSlotsByName: No matching signal for on_tbar_position_valueChanged(int)
qt.core.qmetaobject.connectslotsbyname: QMetaObject::connectSlotsByName: No matching signal for on_actionShowTransitionProperties_triggered()
qt.core.qmetaobject.connectslotsbyname: QMetaObject::connectSlotsByName: No matching signal for on_actionHideTransitionProperties_triggered()
info: OBS 28.1.2-2 (linux)
info: ---------------------------------
info: ---------------------------------
info: audio settings reset:
        samples per sec: 48000
        speakers:        2
        max buffering:   960 milliseconds
        buffering type:  dynamically increasing
info: ---------------------------------
info: Initializing OpenGL...
info: Loading up OpenGL on adapter NVIDIA Corporation NVIDIA GeForce GTX 1060 3GB/PCIe/SSE2
info: OpenGL loaded successfully, version 3.3.0 NVIDIA 525.60.11, shading language 3.30 NVIDIA via Cg compiler
info: ---------------------------------
info: video settings reset:
        base resolution:   1920x1080
        output resolution: 1280x720
        downscale filter:  Bicubic
        fps:               30/1
        format:            NV12
        YUV mode:          Rec. 709/Partial
info: NV12 texture support not available
info: P010 texture support not available
info: Audio monitoring device:
        name: Default
        id: default
info: ---------------------------------
warning: Failed to load 'en-US' text for module: 'decklink-captions.so'
warning: Failed to load 'en-US' text for module: 'decklink-output-ui.so'
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
warning: A DeckLink iterator could not be created.  The DeckLink drivers may not be installed
warning: Failed to initialize module 'decklink.so'
info: [pipewire] Available captures:
info: [pipewire]     - Desktop capture
info: [pipewire]     - Window capture
info: [linux-vkcapture] plugin loaded successfully (version 1.2.2)
info: NVENC supported
info: FFMPEG VAAPI supported
info: [StreamFX] Loading Version 0.12.0b164-ga3a84595
info: [StreamFX] <gfx::opengl> Version 3.3 initialized.
info: [StreamFX] Loaded Version 0.12.0b164-ga3a84595
info: ---------------------------------
info:   Loaded Modules:
info:     StreamFX.so
info:     text-freetype2.so
info:     rtmp-services.so
info:     obs-x264.so
info:     obs-vst.so
info:     obs-transitions.so
info:     obs-outputs.so
info:     obs-libfdk.so
info:     obs-filters.so
info:     obs-ffmpeg.so
info:     linux-vkcapture.so
info:     linux-v4l2.so
info:     linux-pulseaudio.so
info:     linux-pipewire.so
info:     linux-jack.so
info:     linux-capture.so
info:     linux-alsa.so
info:     image-source.so
info:     frontend-tools.so
info:     decklink-output-ui.so
info:     decklink-captions.so
info: ---------------------------------
QLayout: Attempting to add QLayout "" to OBSBasic "OBSBasic", which already has a layout
QWidget::setTabOrder: 'first' and 'second' must be in the same window
info: ==== Startup complete ===============================================
info: All scene data cleared
info: ------------------------------------------------
info: pulse-input: Server name: 'PulseAudio (on PipeWire 0.3.61) 15.0.0'
info: pulse-input: Audio format: s16le, 48000 Hz, 2 channels
info: pulse-input: Started recording from 'alsa_output.usb-Logitech_G_series_G435_Wireless_Gaming_Headset_202105190004-00.iec958-stereo.monitor' (default)
info: [Loaded global audio device]: 'Desktop Audio'
info: pulse-input: Server name: 'PulseAudio (on PipeWire 0.3.61) 15.0.0'
info: pulse-input: Audio format: s32le, 48000 Hz, 8 channels
info: pulse-input: Started recording from 'alsa_output.pci-0000_00_1b.0.pro-output-0.monitor' (default)
info: [Loaded global audio device]: 'Mic/Aux'
info: Switched to scene 'Scene'
info: ------------------------------------------------
info: Loaded scenes:
info: - scene 'Scene':
info: ------------------------------------------------
Illegal instruction (core dumped)
```

Alpha 77 only works until I try to add a plugin from StreamFX and then crashes with the same error:
```
info: User added source 'Image' (image_source) to scene 'Scene'
info: User added filter 'Blur' (streamfx-filter-blur) to source 'Image'
Illegal instruction (core dumped)
```

**Reply by skeletonbow** [reply]
> My clean install of OBS crashes on launch when I use any version of this plugin past alpha 77 with error `Illegal Instruction (core dumped)`. this is my log
> 
> ```
> info: CPU Name: Intel(R) Xeon(R) CPU E5-1620 v2 @ 3.70GHz
...
> Illegal instruction (core dumped)
> ```

This Intel microprocessor no longer meets the instruction set hardware requirements to run newer releases of StreamFX.  You'll have to either keep using older releases of StreamFX, upgrade to a supported processor, or try to find alternative solutions that still support this 10 year old processor.  Hope this helps.

**Reply by Xaymar** [reply]
[Building from Source](https://github.com/Xaymar/obs-StreamFX/wiki/Building) is an option if your System does not meet the requirements set in the Release Notes. Note that self-built binaries recieve no support at all, even for bugs reproducible in a released version.

**Comment by DaeTheDerg** [comment]
> [Building from Source](https://github.com/Xaymar/obs-StreamFX/wiki/Building) is an option if your System does not meet the requirements set in the Release Notes. Note that self-built binaries recieve no support at all, even for bugs reproducible in a released version.

I see
Is there any reason why StreamFX specifically no longer supports the Xeon E5-1620 v2 when other intel processors also released in 2013 are?
I have never had a problem like this with any other programs I've used - even other rendering programs like face tracking software & any modern games - let alone a plugin for a program that works perfectly fine otherwise.
I'm just curious if there is a possibility to return support for sandy bridge CPUs in the future for non-alpha releases without building from source? 

EDIT:
Apologies if my tone sounded combative, that was not my intention!

**Reply by Xaymar** [reply]
This was discussed quite extensively on Discord, but TL;DR: Open Source hobby project primarily made for me, then friends. There are no plans to downgrade from x86-64-v3 to x86-64-v2 or less. If someone doesn't like it, either pay up, or submit a PR, or fork the project.

**Comment by chillinwithdev** [comment]
I'm sure there are a tone of people saying the same thing, but the most recent OBS update 28.1.2 caused my StreamFX to no longer work. When I launch OBS I get the "broken plug-in" warning for StreamFX, and sadly a lot of my OBS scenes relied heavily on the features of StreamFX. Do we have an ETA on when this plug-in will be up and running again? It's been such an amazing tool and not having any updates or info is a bit disappointing. Obviously, I don't know the struggles and tolls involved in creating a plug-in like this, but I hope it gets fixed soon...

**Reply by Xaymar** [reply]
When it is done. If that's not to your liking, you can send in a comission.

**Reply by andlaw** [reply]
the most hated question from Devs.. ETA!  a lot of us are and have been waiting, and if you've been watching since OBS 28.. you will have noticed there have been a lot of updates.  the release process is also posted above, Alpha -> Beta -> Candidate -> Release!  it's in the beta stage... soon! >:)

**Comment by eeproduktionen** [comment]
Workes great for me with OBS 29. My main effect is “blur” - no issues here.

**Comment by Ahmadina91** [comment]
i've got an issue, when installing there is no option for change destination option, its always direct to "programdata", how i fix this?

**Reply by Ahmadina91** [reply]
![image](https://user-images.githubusercontent.com/121887982/210403050-147b24ef-9fb9-4509-9377-1e9e4bada03a.png)

**Reply by Xaymar** [reply]
Explained in the Release Notes, which you could have read instead of writing here.
